src/Xamarin.Android.Build.Tasks/Tasks/JavaDoc.cs

```diff
@@ -53,9 +62,27 @@ protected override string GenerateCommandLineCommands ()
 			cmd.AppendFileNameIfNotNull (context_src);
 			cmd.AppendSwitch ("-subpackages");
 			cmd.AppendSwitch (".");
+			var cps = ReferenceJars?.ToList () ?? new List<string> ();
+			if (JavaPlatformJar != null)
+				cps.Add (JavaPlatformJar);
+			if (cps.Any ()) {
+				if (Environment.OSVersion.Platform != PlatformID.Unix)
+					cmd.AppendSwitch ("-cp " + string.Join (";", cps.Select (cp => '"' + cp + '"')));
```

Comment:
@dellis1972: Should this follow `<CreateMultiDexMainDexClassList/>`/`<Proguard/>` and instead do:

```csharp
var enclosingChar = OS.IsWindows ? "\"" : string.Empty;
cmd.AppendSwitchUnquotedIfNotNull ("-cp ", $"{enclosingChar}'" + string.Join ($"'{Path.PathSeparator}'", libjars.Distinct ()) + $"'{enclosingChar}");
```

Reply:
Is that code even readable? Is Path.PathSeparator always used by JDK?

Reply:
It may be ugly, but it works with the vagaries of Windows vs. Linux path quoting.

Reply:
Let's stop splitting the same argument in two places. This is NOT what you showed in the other comment. It never does `-cp "foo bar:baz"`. Instead it does `-cp 'foo bar':'baz'. And it does not work.